pennylane/devices/default_clifford.py

```diff
@@ -519,58 +686,298 @@ def _measure_state(self, tableau_simulator, circuit, global_phase):
+        wires = kwargs.get("circuit").wires
+        tableau = tableau_simulator.current_inverse_tableau().inverse()
+        z_stabs = qml.math.array([tableau.z_output(wire) for wire in range(len(wires))])
+        return self._measure_stabilizer_entropy(z_stabs, list(meas.wires), meas.log_base)
+
+    def _measure_mutual_info(self, meas, tableau_simulator, **kwargs):
+        """Measure the mutual information between the subsystems of simulator device."""
+        wires = kwargs.get("circuit").wires
+        tableau = tableau_simulator.current_inverse_tableau().inverse()
+        z_stabs = qml.math.array([tableau.z_output(wire) for wire in range(len(wires))])
+        indices0, indices1 = getattr(meas, "_wires")
+        return self._measure_stabilizer_entropy(
+            z_stabs, list(indices0), meas.log_base
+        ) + self._measure_stabilizer_entropy(z_stabs, list(indices1), meas.log_base)
+
+    def _measure_purity(self, meas, tableau_simulator, **kwargs):
+        r"""Measure the purity of the state of simulator device.
+
+        Computes the state purity using the monotonically decreasing second-order Rényi entropy
+        form given in `Sci Rep 13, 4601 (2023) <https://www.nature.com/articles/s41598-023-31273-9>`_.
+        We utilize the fact that Rényi entropies are equal for all Rényi indices ``n`` for the
+        stabilizer states.
 
+        Args:
+            stabilizer (TensorLike): stabilizer set for the system
+            wires (Iterable): wires describing the subsystem
+            log_base (int): base for the logarithm.
+
+        Returns:
+            (float): entanglement entropy of the subsystem
+        """
+        wires = kwargs.get("circuit").wires
+        if wires == meas.wires:
+            return qml.math.array(1.0)
+
+        tableau = tableau_simulator.current_inverse_tableau().inverse()
+        z_stabs = qml.math.array([tableau.z_output(wire) for wire in range(len(wires))])
+        return 2 ** (-self._measure_stabilizer_entropy(z_stabs, list(meas.wires), log_base=2))
+
+    # pylint: disable=protected-access
     @staticmethod
-    def _measure_purity(meas_op, circuit):
-        """Measure the purity of the state of simulator device"""
-        if circuit.wires != meas_op.wires:
-            raise NotImplementedError(
-                "default.clifford doesn't support measuring the purity of a subset of wires at the moment."
+    def _measure_stabilizer_entropy(stabilizer, wires, log_base=None):
+        r"""Computes the Rényi entanglement entropy using stabilizer information.
+
+        Computes the Rényi entanglement entropy :math:`S_A` for a subsytem described by
+        :math:`A`, :math:`S_A = \text{rank}(\text{proj}_A {\mathcal{S}}) - |\mathcal{S}|`,
+        where :math:`\mathcal{S}` is the stabilizer group for the system using the theory
+        described in `arXiv:1901.08092 <https://arxiv.org/abs/1901.08092>`_.
```

Comment:
Didnt know that, pretty cool! Might be worth to make this user-facing somewhere!